File: Python/Leetcode/Easy/Find a Corresponding Node of a Binary Tree in a Clone of That Tree(DFS).py

# 1379. Find a Corresponding Node of a Binary Tree in a Clone of That Tree
# https://leetcode.com/problems/find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree/description/


class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def getTargetCopy(self, original: TreeNode, cloned: TreeNode, target: TreeNode) -> TreeNode:

        stack = [cloned]

        while stack:
            now = stack.pop()

            # 같은 노드 이면 반환
            if now.val == target.val:
                return now

            # 왼쪽 노드 있으면 stack에 추가
            if now.left:
                stack.append(now.left)

            # 오른쪽 노드 있으면 stack에 추가
            if now.right:
                stack.append(now.right)

File: javascript/leetcode/Easy/Find a Corresponding Node of a Binary Tree in a Clone of That Tree(DFS).js
/*
 * 1379. Find a Corresponding Node of a Binary Tree in a Clone of That Tree
 * https://leetcode.com/problems/find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree/description/
 */

function TreeNode(val) {
    this.val = val;
    this.left = this.right = null;
}
/**
 * @param {TreeNode} original
 * @param {TreeNode} cloned
 * @param {TreeNode} target
 * @return {TreeNode}
 */

var getTargetCopy = function(original, cloned, target) {

    let stack = Array();
    stack.push(cloned);

    while(stack){
        const node = stack.pop();
        // 같은 노드 이면 반환
        if(node.val === target.val){
            return node;
        }

        // 왼쪽 노드 있으면 stack에 추가
        if(node.left){
            stack.push(node.left);
        }

        // 왼쪽 노드 있으면 stack에 추가
        if(node.right){
            stack.push(node.right);
        }

    }

};